packages/dts-generator/src/phases/json-fixer.ts

```diff
@@ -904,10 +919,12 @@ export function fixApiJsons(
   targetLibJson: ApiJSON,
   dependencies: ApiJSON[],
   directives: Directives,
+  generateGlobals?: boolean,
 ) {
   // Part 1: "prepare JSON"
   let targetLibFixedJson = _prepareApiJson(targetLibJson, directives, {
     mainLibrary: true,
+    generateGlobals,
```

Comment:
shouldn't the flag be propagated also in the 2nd _prepareApiJson call for the dependencies? As types of the main library might depend on the types from the dependencies, this could matter in future (e.g. if an esmOnly signature in the sap.m directives must match an inherited esmOnly signature in sap.ui.core).

Reply:
Right. Fixed.